Transaction Management Dashboard

jevinjojo opened the pull request:
## Issue 
- Fixes #200
This PR is a follow-up to PR #261.

https://github.com/user-attachments/assets/5f68178d-dc31-49a5-9c6f-9014dc8b396b



## Description

 * Transaction management for the BraidPool Bitcoin mining
 * Table component to render transaction rows
 * Fetch transaction data from the backend (BriadPool API)

jevinjojo committed:
Transactions Page

jevinjojo commented:
@priyashuu i realized i was making changes directly on the main branch, but in dev the dashboard is completely different.

zaidmstrr committed:
fix: stabilize str::from_utf8 by adding std:: prefix (#308)

priyashuu commented:
> @priyashuu i realized i was making changes directly on the main branch, but in dev the dashboard is completely different.

Yes, that's why I was telling you to fetch first from the dev branch and then update with your branch.

priyashuu commented:
From the screen recording, it’s clear that we need to improve the ui Please check the other tab and try to align it with our current UI. I suggest using fewer colors to keep the design simple and basic.

LMK how I can test this, especially regarding how you are fetching the data.
 
A little suggestion: don't add Jest tests for now. Once we have reviewed all your changes, you can include the Jest tests at the end to avoid having to modify the tests each time you make edits.

jevinjojo committed:
manual testing

jevinjojo commented:
> From the screen recording, it’s clear that we need to improve the ui Please check the other tab and try to align it with our current UI. I suggest using fewer colors to keep the design simple and basic.
> 
> LMK how I can test this, especially regarding how you are fetching the data.
> 
> A little suggestion: don't add Jest tests for now. Once we have reviewed all your changes, you can include the Jest tests at the end to avoid having to modify the tests each time you make edits.

hi, I've updated it by adding the testing guide that was written for the backend #259

priyashuu reviewed (commented):
Hey @jevinjojo, thanks for raising the PR!

I reviewed your changes, you added a Transactions tab, but transactions are already implemented inside the `Bitcoin Statistics `tab.

Before adding new components, please go through the entire dev branch to understand the existing structure.

The filter category you created can be added directly to the Transactions section inside Bitcoin Statistics instead of creating a separate tab.

LMK if you need any help navigating the dev branch or integrating this properly.

priyashuu reviewed (commented):
Right now, the Transaction Categories section looks more like documentation.
It’s not necessary to explain everything in such a detailed way on the UI.

Instead of showing large blocks of descriptions, you can simplify this by adding a tooltip that appears when the user selects a category from the filter.
<img width="1757" height="782" alt="image" src="https://github.com/user-attachments/assets/551038db-0587-4360-9591-e4436f4e93fc" />

priyashuu commented:
I suggest reviewing the Bitcoin Statistics page first before adding new changes.
Right now, the issue is that multiple ports were introduced for fetching data, and we’re trying to remove that, we already discussed this on Discord.

If you need any help, feel free to ping us on Discord.

For now, I’m marking this PR as a draft. Once you update your changes, we’ll continue the review.

jevinjojo commented:
> I suggest reviewing the Bitcoin Statistics page first before adding new changes. Right now, the issue is that multiple ports were introduced for fetching data, and we’re trying to remove that, we already discussed this on Discord.
> 
> If you need any help, feel free to ping us on Discord.
> 
> For now, I’m marking this PR as a draft. Once you update your changes, we’ll continue the review.

@priyashuu I will update the changes soon

krishi-agrawal committed:
mobile ui stats revamp (#287)

Signed-off-by: krishi-agrawal <krishi.agrawal26@gmail.com>
Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

krishi-agrawal committed:
dashboard footer revamp (#290)

footer changes

fix ui

Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

s373nZ committed:
refactor: Remove template retry logic in IPC listener. (#312)

Also Updates IPC listener documentation block for `get_template()` via
review feedback from GitHub Copilot.

jevinjojo commented:
@priyashuu 
<img width="1741" height="451" alt="Screenshot 2025-12-22 223845" src="https://github.com/user-attachments/assets/dacf3e71-25e4-472d-b0d5-69ab3b291669" />

I have integrated the transaction filtering into existing table. other few enhancements need to be done, any changes need to apply?

priyashuu commented:
> @priyashuu <img alt="Screenshot 2025-12-22 223845" width="1741" height="451" src="https://private-user-images.githubusercontent.com/176790149/529321846-dacf3e71-25e4-472d-b0d5-69ab3b291669.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY0MjQ3NTIsIm5iZiI6MTc2NjQyNDQ1MiwicGF0aCI6Ii8xNzY3OTAxNDkvNTI5MzIxODQ2LWRhY2YzZTcxLTI1ZTQtNDcyZC1iMGQ1LTY5YWIzYjI5MTY2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyMlQxNzI3MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZWUyYTNiODkyYWY2YzdhMTYyYWI3ZDcyZmQ4YzE5N2QxMTUxMTk3ZTNjMmQwMWU3MGRiMTkzOTIyNzYyZGY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vvOxL3ezkfBe7HL4k-aumEmLwIPpbn7Qr5Euxzrmx2U">
> 
> I have integrated the transaction filtering into existing table. other few enhancements need to be done, any changes need to apply?

I’ll review it first and get back to you.
In the meantime, could you please add a screen recording? It would be really helpful.

priyashuu commented:
> @priyashuu <img alt="Screenshot 2025-12-22 223845" width="1741" height="451" src="https://private-user-images.githubusercontent.com/176790149/529321846-dacf3e71-25e4-472d-b0d5-69ab3b291669.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY3NTcyMDEsIm5iZiI6MTc2Njc1NjkwMSwicGF0aCI6Ii8xNzY3OTAxNDkvNTI5MzIxODQ2LWRhY2YzZTcxLTI1ZTQtNDcyZC1iMGQ1LTY5YWIzYjI5MTY2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIyNlQxMzQ4MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZDg3YTI0MTBlNWQ1MDIwM2Y2M2I4ODdhNTc5OTNjMmY5ZGMxYWU1OTU5N2VkYTZmMjMyNzgwYzBmYWNkMzkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-lwp7WW0h7_AeJ_L9rQq_IoWIwBgQw-LLiKsulPlhqA">
> 
> I have integrated the transaction filtering into existing table. other few enhancements need to be done, any changes need to apply?

Please push your latest changes so that I can review them.

krishi-agrawal committed:
Standardised Loader Screens for a Consistent UI (#315)

* unify loader screens

* added mapping

---------

Co-authored-by: Bob McElrath <mcelrath@users.noreply.github.com>

jevinjojo committed:
changes

jevinjojo commented:
@priyashuu Sorry for the late reply.  I have pushed the changes


https://github.com/user-attachments/assets/3d7fe4bb-82c4-4180-a571-8b97ce9212c5

jevinjojo committed:
components and utilities changes

jevinjojo committed:
duplicate

priyashuu reviewed (commented):
I haven’t reviewed the code yet and will do so shortly. In the meantime, kindly address the CI test failures.

jevinjojo committed:
test failures and formatting

priyashuu commented:
Hi Jevin, are you currently working on this pr?

jevinjojo commented:
Hi @priyashuu not right now, it would be great if someone could take up both. Thanks

priyashuu commented:
Thank you for your efforts. I am closing this PR as no recent activity or changes have been made.
